src/libstd/sys/common/net2.rs

```diff
@@ -268,6 +269,16 @@ impl FromInner<Socket> for TcpStream {
     }
 }
 
+impl fmt::Debug for TcpStream {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.debug_struct("TcpStream")
+        .field("addr", &self.socket_addr())
+        .field("peer", &self.peer_addr())
```

Comment:
Could these fields be omitted entirely if an error is encountered? It may not be super useful to include `Err(Error { repr: Os(9) })` in the output unfortunately.